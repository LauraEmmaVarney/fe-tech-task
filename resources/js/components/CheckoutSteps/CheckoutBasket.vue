<template>
    <base-card>
        <template v-slot:header>
            <div class="row align-items-center">
                <div class="col-auto">
                    <i class="fal fa-shopping-basket text-primary fa-2x"/>
                </div>
                <div class="col pl-0">
                    <h5 class="mb-0 font-weight-normal">Your Items</h5>
                </div>
            </div>
        </template>
        <template v-slot:body>
            <table class="table small mb-0 table-cart">
                <thead class="thead-light">
                    <tr>
                        <th>Item</th>
                        <th>Price</th>
                        <th class="text-right">Quantity</th>
                        <th class="text-right">Total</th>
                    </tr>
                </thead>
                <tbody>
                    <tr 
                        v-for="(item, index) in items"
                        :key="index"
                        class="row-product"
                    >
                        <td>{{ item.name }}</td>
                        <td>${{ item.price }}</td>
                        <td class="text-right">{{ item.qty }}</td>
                        <td class="text-right">{{ store.currencySymbol }}{{ item.rowprice }}</td>
                    </tr>
                    <tr class="border-top border-1">
                        <td></td>
                        <td></td>
                        <td class="text-right">Total:</td>
                        <td class="text-right">$ {{basketTotal}}</td>
                    </tr>
                </tbody>
            </table>
        </template>
    </base-card>
</template>

<script>

import BaseCard from "../Utilities/BaseCard.vue";

    export default {
        name: "CheckoutBasket",
        components: {
            BaseCard
        },
        props: {
            items: {
                type: Array,
            },
            store: {
                type: Object,
                required: true,
            }
        },
        computed: {
            basketTotal: function() {
                let total = 0;
                for (let i in this.items) {
                    total += this.items[i].rowprice;
                }
                return total;
            }
        },
    }

</script>
