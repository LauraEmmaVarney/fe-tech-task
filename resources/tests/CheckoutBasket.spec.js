import { shallowMount } from '@vue/test-utils'
import CheckoutBasket from '../js/components/CheckoutSteps/CheckoutBasket.vue';

describe('CheckoutBasket component tests', () => {
  const wrapper = shallowMount(CheckoutBasket, {
    propsData: {
      store: {
        "currencySymbol":"$",
        "name":"Test Store",
        "logoUrl": null
      },
      items: [{"id":1,"name":"Test Product","price":5,"qty":1,"rowprice":5},{"id":6,"name":"Test Product 2","price":1.45,"qty":2,"rowprice":2.9}]
    }
  });

  it('mounts the component', () => {
    expect(wrapper.exists()).toBe(true);
  });

  it('find child components by name', () => {
    const childComponent = wrapper.findComponent({ name: 'BaseCard' });
    expect(childComponent.exists()).toBe(true);
  });

  it('checks props data', () => {
    expect(wrapper.props().store).toEqual({
        "currencySymbol":"$",
        "name":"Test Store",
        "logoUrl": null
    });
    expect(wrapper.props().items).toEqual([{"id":1,"name":"Test Product","price":5,"qty":1,"rowprice":5},{"id":6,"name":"Test Product 2","price":1.45,"qty":2,"rowprice":2.9}]);
  });

  it('checks computed properties', () => {
    expect(wrapper.vm.basketTotal).toBe(7.9);
  });

  it('check title', () => {
    const title = wrapper.find('h5');
    expect(title.text()).toBe('Your Items');
  });

  it('check against jest snapshot test', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});