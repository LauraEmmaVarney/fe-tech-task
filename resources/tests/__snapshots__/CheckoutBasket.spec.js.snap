// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutBasket component tests check against jest snapshot test 1`] = `
<base-card-stub>
  <div
    class="row align-items-center"
  >
    <div
      class="col-auto"
    >
      <i
        class="fal fa-shopping-basket text-primary fa-2x"
      />
    </div>
     
    <div
      class="col pl-0"
    >
      <h5
        class="mb-0 font-weight-normal"
      >
        Your Items
      </h5>
    </div>
  </div>
  <table
    class="table small mb-0 table-cart"
  >
    <thead
      class="thead-light"
    >
      <tr>
        <th>
          Item
        </th>
         
        <th>
          Price
        </th>
         
        <th
          class="text-right"
        >
          Quantity
        </th>
         
        <th
          class="text-right"
        >
          Total
        </th>
      </tr>
    </thead>
     
    <tbody>
      <tr
        class="row-product"
      >
        <td>
          Test Product
        </td>
         
        <td>
          $5
        </td>
         
        <td
          class="text-right"
        >
          1
        </td>
         
        <td
          class="text-right"
        >
          $5
        </td>
      </tr>
      <tr
        class="row-product"
      >
        <td>
          Test Product 2
        </td>
         
        <td>
          $1.45
        </td>
         
        <td
          class="text-right"
        >
          2
        </td>
         
        <td
          class="text-right"
        >
          $2.9
        </td>
      </tr>
       
      <tr
        class="border-top border-1"
      >
        <td />
         
        <td />
         
        <td
          class="text-right"
        >
          Total:
        </td>
         
        <td
          class="text-right"
        >
          $ 7.9
        </td>
      </tr>
    </tbody>
  </table>
</base-card-stub>
`;
